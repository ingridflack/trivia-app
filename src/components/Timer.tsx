import { CircularProgress, CircularProgressLabel } from "@chakra-ui/react";

interface TimerProps {
  value: number;
  maxValue: number;
}

export default function Timer({ value, maxValue }: TimerProps) {
  return (
    <CircularProgress
      value={value}
      color="green.400"
      size="120px"
      min={0}
      max={maxValue}
    >
      <CircularProgressLabel>{value}</CircularProgressLabel>
    </CircularProgress>
  );
}
