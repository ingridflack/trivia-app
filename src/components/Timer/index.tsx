import { CircularProgress, CircularProgressLabel } from "@chakra-ui/react";

interface TimerProps {
  value: number;
  maxValue: number;
}

export default function Timer({ value, maxValue }: TimerProps) {
  const color = value <= 5 ? "red.400" : "green.400";

  return (
    <CircularProgress
      value={value}
      color={color}
      size="74px"
      display="flex"
      alignItems="center"
      justifyContent="center"
      min={0}
      max={maxValue}
      thickness="4px"
    >
      <CircularProgressLabel color={color} fontWeight={700}>
        {value}
      </CircularProgressLabel>
    </CircularProgress>
  );
}
