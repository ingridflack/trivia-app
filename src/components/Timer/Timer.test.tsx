import { cleanup, render } from "@testing-library/react";
import { afterEach, describe, expect, it } from "vitest";
import Timer from ".";
import { ChakraProvider } from "@chakra-ui/react";

const renderWithChakraProvider = (ui: React.ReactElement) => {
  return render(<ChakraProvider>{ui}</ChakraProvider>);
};

describe("Timer", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match the snapshot", () => {
    const { container } = render(<Timer value={10} maxValue={20} />);
    expect(container).toMatchSnapshot();
  });

  it("should render with value and maxValue props", () => {
    const { getByText } = renderWithChakraProvider(
      <Timer value={10} maxValue={100} />
    );

    const progressLabel = getByText("10");
    expect(progressLabel).toBeTruthy();
  });
});
