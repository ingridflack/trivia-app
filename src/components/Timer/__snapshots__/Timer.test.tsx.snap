// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Timer > should match the snapshot 1`] = `
<div>
  <div
    aria-valuemax="20"
    aria-valuemin="0"
    aria-valuenow="10"
    class="chakra-progress css-1wsiu6z"
    role="progressbar"
  >
    <svg
      class="css-1hrl48y"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-1bhxzgy"
        cx="50"
        cy="50"
        r="42"
        stroke-width="4px"
      />
      <circle
        class="chakra-progress__indicator css-12qt19m"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="132 132"
        stroke-dashoffset="66"
        stroke-width="4px"
      />
    </svg>
    <div
      class="css-tvcbta"
    >
      10
    </div>
  </div>
</div>
`;
